src/Microsoft.VisualStudio.Threading/AsyncReaderWriterResourceLock`2.cs

```diff
@@ -697,7 +697,7 @@ internal async Task<TResource> GetResourceAsync(TMoniker resourceMoniker, Cancel
         {
             using (AsyncReaderWriterResourceLock<TMoniker, TResource>.ResourceReleaser resourceLock = this.AcquirePreexistingLockOrThrow())
             {
-                TResource? resource = await this.service.GetResourceAsync(resourceMoniker, cancellationToken).ConfigureAwait(false);
+                TResource? resource = await this.service.GetResourceAsync(resourceMoniker, cancellationToken).ConfigureAwaitRunInline();
```

Comment:
I guess the net effect of this change is to force it to inline in more scenarios than before. Perhaps while on the UI thread? 

The next thing we do here is enter a lock. If the lock is contested, we'll block the antecedent task from invoking any other continuations. But more importantly perhaps, if we're on the UI thread we'll pump messages, which for lock or scheduling related code tends to be _really_ bad. Perhaps we should add a no-pumping using block around the lock.

Reply:
This is to try to reduce the chance to reschedule the task continuation unnecessarily. I don't think it really changes the behavior here, because  it would be the same if GetResourceAsync doesn't yield, which is actually the most common situation in the product, as the project evaluation lazy would be done except the narrow initial loading time. If a no-puming logic is required, it would be required anyway, independent to this change.  (The contention should be unlikely a problem because inner code is really to register a few tasks, instead of doing any real work. Unlike the JTF, the sync lock in the lock service never shows to be a major contention point in any trace.